### PR TITLE
Use @sap-cloud-sdk/cli in SAP Cloud SDK for JavaScript tutorials

### PR DESCRIPTION
**Pull request by mr-flannery**
@sacnl 
They're not perfect yet, but they work and we can start positioning the CLI. I suggest we do a round of feedback in the new year and look into how complex support for plain JS would be in the `init` command (you'll notice that I removed the TS/JS toggles for now).

### CODE REVIEW

**Comment by marikaner** [comment]
Adding myself in case anything comes up during the holiday time.

File: tutorials/s4sdkjs-getting-started/s4sdkjs-getting-started.md [hunk 1]
```diff
@@ -19,34 +19,36 @@ primary_tag: products>sap-s-4hana-cloud-sdk
 
 [ACCORDION-BEGIN [Step 1: ](Scaffold an application)]
 
-Personally, we are fans of `TypeScript` and recommend using it for most applications. However, we also offer a version of our application scaffolding for plain `JavaScript`. If a step in a tutorial contains instructions or code that is specific to `TypeScript` or `JavaScript`, you can toggle between whatever you prefer at the top of each step.
+Personally, we are fans of `TypeScript` and recommend using it for most applications. However, if you prefer using plain `JavaScript`, we recommend looking at step 4. If a step in a tutorial contains instructions or code that is specific to `TypeScript` or `JavaScript`, you can toggle between whatever you prefer at the top of each step.
```
Comment (sacnl) [inline tutorials/s4sdkjs-getting-started/s4sdkjs-getting-started.md]: > you can toggle between whatever you prefer at the top of each step

Should be removed as well.

File: tutorials/s4sdkjs-getting-started/s4sdkjs-getting-started.md [hunk 2]
```diff
@@ -19,34 +19,36 @@ primary_tag: products>sap-s-4hana-cloud-sdk
...
 
 ```Shell
-git clone --single-branch --branch scaffolding-js --origin scaffolding https://github.com/SAP/cloud-s4-sdk-examples.git <path/to/your/project>
+sap-cloud-sdk init my-sdk-project
 ```
 
-[OPTION END]
-
-and enter the freshly cloned project:
+If your folder is empty, this will create a new `nest.js` application from scratch. During the process, the CLI will ask to you provide some details, like your project's name. Since this will already install all the necessary dependencies, this might take a minute. If everything worked correctly, you should see output like this:
```
Comment (sacnl) [inline tutorials/s4sdkjs-getting-started/s4sdkjs-getting-started.md]: Mention to answer "Should a new `nest.js` project be initialized in this folder?" with `y`, and accept the suggested project name.

File: tutorials/s4sdkjs-getting-started/s4sdkjs-getting-started.md [hunk 3]
```diff
@@ -19,34 +19,36 @@ primary_tag: products>sap-s-4hana-cloud-sdk
...
-
-[OPTION BEGIN [JavaScript]]
-
-To create an `Express.js` application that already contains all the files and configuration you need to use the SAP Cloud SDK for JavaScript, simply clone our JavaScript scaffolding application as follows.
+This will install the CLI globally on your machine, allowing you to use it anywhere.
+Now you can create a new project by running the CLI's `init` command:
 
 ```Shell
-git clone --single-branch --branch scaffolding-js --origin scaffolding https://github.com/SAP/cloud-s4-sdk-examples.git <path/to/your/project>
+sap-cloud-sdk init my-sdk-project
```
Comment (sacnl) [inline tutorials/s4sdkjs-getting-started/s4sdkjs-getting-started.md]: This creates a new folder `my-sdk-project`, so is it still important to run this in an empty folder?

File: tutorials/s4sdkjs-getting-started/s4sdkjs-getting-started.md [hunk 4]
```diff
@@ -58,12 +60,13 @@ The project contains the following files and folders, among others, to get you s
 
 ### NPM / Project
 
-- **`package.json`**: Specifies dependencies, metadata and user-defined scripts. The provided scaffolding comes with some predefined scripts and dependencies, that will be explained in detail in the course of this group of tutorials.
+- **`package.json`**: Specifies dependencies, metadata and user-defined scripts. The application comes with some predefined scripts and dependencies, that will be explained in detail in the course of this group of tutorials.
 - **`.npmrc`**: The **`npm`** configuration file. In the scaffolding we specify the registry for the `@sap` scope, where the `SAP Cloud SDK` libraries are published.
 
 ### TypeScript
 
 - **`tsconfig.json`**: Configuration file for `TypeScript`. This is not present in the plain `JavaScript` version.
```
Comment (sacnl) [inline tutorials/s4sdkjs-getting-started/s4sdkjs-getting-started.md]: "not present" -> "not needed"

File: tutorials/s4sdkjs-getting-started/s4sdkjs-getting-started.md (hunk 2 above)
Comment (sacnl) [reply tutorials/s4sdkjs-getting-started/s4sdkjs-getting-started.md]: Mention that this may take a while for npm install etc

File: tutorials/s4sdkjs-getting-started/s4sdkjs-getting-started.md [hunk 5]
```diff
@@ -58,12 +60,13 @@ The project contains the following files and folders, among others, to get you s
...
+- **`tslint.json`**: Configuration file for `tslint`, the de facto default linter for `TypeScript`.
 
 ### Continuous Delivery
```
Comment (sacnl) [inline tutorials/s4sdkjs-getting-started/s4sdkjs-getting-started.md]: Folder cx-server is no longer being created -> remove

File: tutorials/s4sdkjs-getting-started/s4sdkjs-getting-started.md [hunk 6]
```diff
@@ -78,46 +81,51 @@ The project contains the following files and folders, among others, to get you s
 ### Local development
 
 - **`initialize.js`**: Script for initial setup.
```
Comment (sacnl) [inline tutorials/s4sdkjs-getting-started/s4sdkjs-getting-started.md]: Not generated

File: tutorials/s4sdkjs-odata-service-cloud-foundry/s4sdkjs-odata-service-cloud-foundry.md [hunk 7]
```diff
@@ -29,117 +29,59 @@ Alternatively, many APIs can also be tested using the sandbox of the SAP API Bus
 
 [ACCORDION-BEGIN [Step 2: ](Add a custom route)]
 
-[OPTION BEGIN [TypeScript]]
+Initially, the app only contains the `index` and `hello-world` routes. We will add another route for `business-parters` that will list all available business partners.
```
Comment (sacnl) [inline tutorials/s4sdkjs-odata-service-cloud-foundry/s4sdkjs-odata-service-cloud-foundry.md]: Fix: `business-partners`

File: tutorials/s4sdkjs-odata-service-cloud-foundry/s4sdkjs-odata-service-cloud-foundry.md [hunk 8]
```diff
@@ -29,117 +29,59 @@ Alternatively, many APIs can also be tested using the sandbox of the SAP API Bus
...
+In order to use the `SAP Cloud SDK for JavaScript` to make a call to an `OData` service, add the `virtual data model` (`VDM`) for this service to your dependencies. For this tutorial we are using the `VDM` for the business partner service. Install it with the following command:
 
 ```Shell
 npm install @sap/cloud-sdk-vdm-business-partner-service
 ```
 
-[OPTION BEGIN [TypeScript]]
-
-Import the entity you want to make a call to into your application. In this tutorial we are importing the business partner entity of the business partner service. Add the following line to the top of the `business-partner-route.ts`.
+Import the entity you want to make a call to into your application. In this tutorial we are importing the business partner entity of the business partner service. Add the following line to the top of the `business-partner.constroller.ts`.
```
Comment (sacnl) [inline tutorials/s4sdkjs-odata-service-cloud-foundry/s4sdkjs-odata-service-cloud-foundry.md]: Fix: `business-partner.controller.ts`

File: tutorials/s4sdkjs-odata-service-cloud-foundry/s4sdkjs-odata-service-cloud-foundry.md [hunk 9]
```diff
@@ -189,115 +129,42 @@ return BusinessPartner.requestBuilder()
...
-function getAllBusinessPartners() {
-  return BusinessPartner.requestBuilder()
-    .getAll()
-    .execute({
-      url: 'http://localhost:3000'
+@Get('business-partners')
+getBusinessPartners() {
+  return getAllBusinessPartners()
+    .catch(error => {
+      throw new HttpException(`Failed to get business partners - ${error.message}`, 500);
```
Comment (sacnl) [inline tutorials/s4sdkjs-odata-service-cloud-foundry/s4sdkjs-odata-service-cloud-foundry.md]: Needs import.

File: tutorials/s4sdkjs-odata-service-cloud-foundry/s4sdkjs-odata-service-cloud-foundry.md [hunk 10]
```diff
@@ -306,8 +173,6 @@ Now restart your server and reload the `http://localhost:8080/business-partners`
```
Comment (sacnl) [inline tutorials/s4sdkjs-odata-service-cloud-foundry/s4sdkjs-odata-service-cloud-foundry.md]: Remove "now restart your server"

**Comment by sacnl** [comment]
The section "Step 5: Manage destinations centrally (optional)" needs an update.
The nest way of doing this would be (see [docs](https://docs.nestjs.com/techniques/configuration)):
```
npm i --save @nestjs/config

import { Module } from '@nestjs/common';
import { ConfigModule } from '@nestjs/config';
import { AppController } from './app.controller';
import { AppService } from './app.service';
import { BusinessPartnerController } from './business-partner.controller';

@Module({
  imports: [ConfigModule.forRoot()],
  controllers: [AppController, BusinessPartnerController],
  providers: [AppService],
})
export class AppModule {}
```

No install of dotenv necessary, no change to scripts necessary, content of `.env` file remains as is

File: tutorials/s4sdkjs-getting-started/s4sdkjs-getting-started.md (hunk 3 above)
Comment (mr-flannery) [reply tutorials/s4sdkjs-getting-started/s4sdkjs-getting-started.md]: Is this question related to the phrasing "If your folder is empty"? I agree that's not strictly necessary. Maybe rephrase as "Since the target folder is empty"?

Comment (sacnl) [reply tutorials/s4sdkjs-getting-started/s4sdkjs-getting-started.md]: yes

**Comment by sacnl** [comment]
Regarding tutorial "Deploy application to Cloud Foundry"

The warning about ci-package not working on Windows can possibly be removed (although there is a different issue still happening for me (see issue #872 in our backlog).

Not related to the introduction of the CLI:
* there are now many more regions than the two explicitly mentioned - maybe we should rather give instructions where to find the API URL in the cockpit (subaccount overview page)
* why not use 'MockServer' as name of destination - that way, people that did the optional step in a previous tutorial do not have to adapt their code.

**Comment by mr-flannery** [comment]
Good suggestions! I added that to the tuorial. Do you have more feedback? Otherwise I'll open a new PR and tag you there.

**Comment by sacnl** [comment]
That's it - for the moment ;)
